validation.go

```diff
@@ -149,3 +192,79 @@ func convertValidations(validations []Validation) []cpluginv1.ParameterValidatio
 	}
 	return out
 }
+
+// ApplyConfigValidations a utility function that applies all the validations for parameters, return an error that
+// consists of a combination of errors from the configurations.
+func applyConfigValidations(params map[string]Parameter, config map[string]string) error {
+	// cfg is the config map with default values assigned
+	cfg, multiErr := assignParamDefaults(params, config)
+
+	for pKey, param := range params {
+		if len(strings.TrimSpace(cfg[pKey])) != 0 {
+			multiErr = multierr.Append(multiErr, validateParamType(param, pKey, cfg[pKey]))
+		}
+		multiErr = multierr.Append(multiErr, validateParamValue(pKey, cfg[pKey], param))
+	}
+	return multiErr
+}
+
+// validateParamValue validates that a configuration value matches all the validations required for the parameter.
+func validateParamValue(key string, value string, param Parameter) error {
+	var multiErr error
+	var err error
+
+	for _, v := range param.Validations {
+		err = v.validate(value)
+		if err != nil {
+			multiErr = multierr.Append(multiErr, fmt.Errorf("error validating %q: %w", key, err))
+		}
+	}
+	return multiErr
+}
+
+// assignParamDefaults fills any empty configuration with its assigned default value
+// returns an error if a parameter is not recognized.
+func assignParamDefaults(params map[string]Parameter, config map[string]string) (map[string]string, error) {
+	output := make(map[string]string)
+	var multiErr error
+	for key, val := range config {
+		if _, ok := params[key]; !ok {
+			multiErr = multierr.Append(multiErr, fmt.Errorf("unrecognized parameter %q", key))
+			continue
+		}
+		output[key] = val
+		if len(strings.TrimSpace(val)) == 0 {
+			output[key] = params[key].Default
+		}
+	}
+	return output, multiErr
+}
+
+// validateParamType validates that a parameter value is parsable to its assigned type.
+func validateParamType(p Parameter, key string, value string) error {
+	switch p.Type {
+	case ParameterTypeInt:
+		_, err := strconv.Atoi(value)
+		if err != nil {
+			return fmt.Errorf("%q: %q value is not an integer: %w", key, value, ErrInvalidParamType)
```

Comment:
`validateParamValue` will create errors that are prefixed with `error validating %q: ...`, I think we should do the same here if the type is not correct.